src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp

```diff
@@ -55,6 +73,100 @@ void GraphingCalculator::ViewModel::set(GraphingCalculatorViewModel^ vm)
     }
 }
 
+void CalculatorApp::GraphingCalculator::OnShareClick(Platform::Object^ sender, Windows::UI::Xaml::RoutedEventArgs^ e)
+{
+    // Ask the OS to start a share action.
+    if (TheGrapher != nullptr)
+    {
+        DataTransferManager::ShowShareUI();
+    }
+
+}
+
+// When share is invoked (by the user or programatically) the event handler we registered will be called to populate the datapackage with the
+// data to be shared.  We will request the current graph image from teh grapher as a stream that will pass to the share request.
+void GraphingCalculator::OnDataRequested(DataTransferManager^ sender, DataRequestedEventArgs^ args)
+{
+    try
+    {
+        // Get our title from the localized resources
+        auto resourceLoader = Windows::ApplicationModel::Resources::ResourceLoader::GetForCurrentView();
+        auto EmptyEquationString = resourceLoader->GetString(L"EmptyEquationString");
+
+        auto htmlResources = ref new Map<String^, RandomAccessStreamReference^>();
+        std::wstring rawHtml = L"<p><img src='graph.png'></p>";
+
+        auto equations = ViewModel->Equations;
+        rawHtml += L"<p><table cellpadding=\"10\">";
+        rawHtml += L"<col width=\"60\">";
+        rawHtml += L"<row height=\"60\">";
+        for (unsigned i = 0; i < equations->Size; i++)
+        {
+            auto expression = equations->GetAt(i)->Expression->Data();
+            auto color = equations->GetAt(i)->LineColor;
+
+            if (equations->GetAt(i)->Expression->Length() == 0)
+            {
+                expression = EmptyEquationString->Data();
+            }
+
+            rawHtml += L"<tr>";
+
+            rawHtml += L"<td style=\"background-color:rgb(";
+            rawHtml += color.R.ToString()->Data();
+            rawHtml += L",";
+            rawHtml += color.G.ToString()->Data();
+            rawHtml += L",";
+            rawHtml += color.B.ToString()->Data();
+            rawHtml += L"); \">";
+            rawHtml += L"&#402;(&#215;)";
+            rawHtml += L"</td>";
+            rawHtml += L"<td>";
+            rawHtml += expression;
+            rawHtml += L"</td>";
+
+            rawHtml += L"</tr>";
+        }
+        rawHtml += L"</table></p>";
+
+        // Shortcut to the request data
+        auto requestData = args->Request->Data;
+
+        DataPackage^ dataPackage = ref new DataPackage();
+        auto html = HtmlFormatHelper::CreateHtmlFormat(ref new String(rawHtml.c_str()));
+
+        auto titleString = resourceLoader->GetString(L"ShareActionTitle");
+        requestData->Properties->Title = titleString;
+
+        requestData->SetHtmlFormat(html);
```

Comment:
Sharing via outlook desktop shows the raw html. I wonder if we should also include the bmp as an attachment in cases like these? Maybe the [setdataprovider api](https://docs.microsoft.com/en-us/uwp/api/windows.applicationmodel.datatransfer.datapackage.setdataprovider) could help?

Reply:
Updated to add the stream to thumbnail and bitmap.  Let the share target figure out what it wants to use.